sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1199,13 +1199,22 @@ class GpuRegExpExtractMeta(
       case _ =>
     }
 
+    val groupIdx = expr.idx match {
+      case Literal(value, DataTypes.IntegerType) =>
+        Some(value.asInstanceOf[Int])
+      case _ =>
+        willNotWorkOnGpu("GPU only supports literal index")
+        None
+    }
```

Comment:
minor nit: this logic is duplicated and could be refactored into a helper method